packages/cli/src/run-test.ts

```diff
@@ -8,9 +8,24 @@ import * as query from './query';
 import { StreamingFormatter } from './format-helpers';
 
 export function* runTest(config: ProjectOptions, formatter: StreamingFormatter): Operation<void> {
-  let client: Client = yield Client.create(`ws://localhost:${config.port}`);
+
+  let client: Client;
+  let uri = `ws://localhost:${config.port}`;
+
+  try {
+    client = yield Client.create(uri);
+  } catch (e) {
+    if (e.message.includes('websocket server closed connection unexpectedly')) {
+      throw new MainError({ 
+        exitCode: 1,
+        message: `Could not connect to BigTest server on ${uri}. Run "bigtest server" to start the server.`
+      });
+    }
+    throw e;
+  }
```

Comment:
We can wrap the client creation into a single operation, that way no need to re-assign the `client` reference.

```suggestion
  let uri = `ws://localhost:${config.port}`;
  
  let client: Client = yield function*() {
    try {
      return yield Client.create(uri);
    } catch (e) {
      if (e.message.includes('websocket server closed connection unexpectedly')) {
        throw new MainError({ 
          exitCode: 1,
          message: `Could not connect to BigTest server on ${uri}. Run "bigtest server" to start the server.`
        });
      }
      throw e;
    }  
  };
```